applications: nrf_desktop: Document nRF21540EK support (#7665)
nrfconnect/sdk-nrf: applications/nrf_desktop/README.rst

    If a nRF Desktop device supports remote wakeup, the USB connected device goes to suspended state when USB is suspended.
    The device can then trigger remote wakeup of the connected host on user input.

Configuration
*************

|config|

Adding nRF21540 EK shield support
=================================

The nRF Desktop application can be used with the :ref:`ug_radio_fem_nrf21540_ek` shield, an RF front-end module (FEM) for the 2.4 GHz range extension.

> Any nRF Desktop HID application can be built with the shield support by passing the ``SHIELD`` parameter to the build command.

Reviewer: Should we refer only to development kits here? The custom HW (e.g. mouse, kbd, dongle) cannot be used with shield.

Reviewer: `nRF Desktop HID application configured for a development kit can be built ...`

Reviewer: The shield cannot be used out of the box together with nRF desktop's custom hardware (like gaming mouse, keyboard and dongle)

Author: How about:
```
This, out of the box, method of enabling SHIELD support is only available for boards that are fitted with Arduino compatible connector.
```

Reviewer 2: I suggest combining both:
```suggestion
The shield can be used with any nRF Desktop HID application configured for a development kit that is fitted with Arduino-compatible connector (see the :guilabel:`DK` tab in `Requirements`_).
This means that the shield support is not available for nRF Desktop's dedicated , such as ``nrf52840gmouse_nrf52840``, ``nrf52kbd_nrf52832``, or ``nrf52840dongle_nrf52840``.
To build the application with the shield support, pass the ``SHIELD`` parameter to the build command.
```

> For detailed information about building the nRF Desktop application for the nRF21540 EK, see the :ref:`ug_radio_fem_nrf21540_ek_programming` section on the EK documentation page.

Reviewer 3: ```suggestion
For detailed information about building the nRF Desktop application for the nRF21540 EK, see the :ref:`ug_radio_fem_nrf21540_ek_programming` section on the nRF21540 EK documentation page.
```

.. note::

>    For the multi-core build, use the ``hci_rpmsg_`` parameter.

Reviewer: parameter -> child image name ?
Maybe even: `use the ``hci_rpmsg_`` child image name, because in nRF Desktop network core runs ``hci_rpmsg``

Reviewer 2: Sounds good to me.

Author: Done

    If a nRF Desktop device supports remote wakeup, the USB connected device goes to suspended state when USB is suspended.
    The device can then trigger remote wakeup of the connected host on user input.

Configuration
*************

|config|

Adding nRF21540 EK shield support
=================================

The nRF Desktop application can be used with the :ref:`ug_radio_fem_nrf21540_ek` shield, an RF front-end module (FEM) for the 2.4 GHz range extension.
The shield can be used with any nRF Desktop HID application configured for a development kit that is fitted with Arduino-compatible connector (see the :guilabel:`DK` tab in `Requirements`_).
This means that the shield support is not available for nRF Desktop's dedicated boards, such as ``nrf52840gmouse_nrf52840``, ``nrf52kbd_nrf52832``, or ``nrf52840dongle_nrf52840``.
To build the application with the shield support, pass the ``SHIELD`` parameter to the build command.

For detailed information about building the nRF Desktop application for the nRF21540 EK, see the :ref:`ug_radio_fem_nrf21540_ek_programming` section on the nRF21540 EK documentation page.

.. note::

>    For the multi-core build, use the hci_rpmsg_ child image name, because in nRF Desktop network core runs ``hci_rpmsg``.

Reviewer 3: ```suggestion
   For the multi-core build, use the ``hci_rpmsg`` child image name, because in nRF Desktop network core runs ``hci_rpmsg``.
```

>    For the multi-core build, use the hci_rpmsg_ child image name, because in nRF Desktop network core runs ``hci_rpmsg``.

Author: Isn't the problem in the end of the sentence?

>    For the multi-core build, use the hci_rpmsg_ child image name, because in nRF Desktop network core runs ``hci_rpmsg``.

Reviewer 2: Nope, it's about the _

>    For the multi-core build, use the hci_rpmsg_ child image name, because in nRF Desktop network core runs ``hci_rpmsg_``.

Reviewer 3: ```suggestion
   For the multi-core build, use the ``hci_rpmsg_`` ``*childImageName*`` parameter, because in nRF Desktop application, network core runs using ``hci_rpmsg_``.
```
Edited my suggestion.

>    For the multi-core build, use the ``hci_rpmsg_`` as the *childImageName* parameter, because in the nRF Desktop application, network core runs using``hci_rpmsg_``.

Reviewer 3: ```suggestion
   For the multi-core build, use the ``hci_rpmsg_`` as the *childImageName* parameter, because in the nRF Desktop application, network core runs using ``hci_rpmsg_``.
```
My bad, missed a space here.

    If a nRF Desktop device supports remote wakeup, the USB connected device goes to suspended state when USB is suspended.
    The device can then trigger remote wakeup of the connected host on user input.

Configuration
*************

|config|

Adding nRF21540 EK shield support
=================================

The nRF Desktop application can be used with the :ref:`ug_radio_fem_nrf21540_ek` shield, an RF front-end module (FEM) for the 2.4 GHz range extension.

> Any nRF Desktop HID application can be built with the shield support by passing the ``SHIELD`` parameter to the built command.

Reviewer 2: ```suggestion
Any nRF Desktop HID application can be built with the shield support by passing the ``SHIELD`` parameter to the build command.
```
Sorry, missed a typo...

Author: Already fixed

.. note::
    Information about the known issues in nRF Desktop can be found in |NCS|'s :ref:`release_notes` and on the :ref:`known_issues` page.

> Adding nRF21540EK shield support

Reviewer: I think we should add a link here so that user could get more information about the `nRF21540EK`:
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_radio_fem.html#nrf21540-ek

Reviewer: Keep in mind that all of the configuration needed to add shield to the build is described here. Maybe we could even state that we use `hci_rpmsg` child image on netcore in case of nRF53 and just link to the required configuration described there to simplify maintenance?

Author: Linked by @greg-fer . I like this solution.

> Adding nRF21540EK shield support

Reviewer: I think adding it under configuration would be nice. We could even explicitly state that this is optional step.

> Adding nRF21540EK shield support

Author: Already done by @greg-fer

================================

The nRF21540EK contains RF front-end module (FEM) for 2.4 GHz range extension.
Any application can be built with the shield support that was already implemented
just passing the SHIELD parameter to the built command.

The build command for nRF21540EK would look differently for single and multicore configuration.

> Single core application

Reviewer 3: ```suggestion
Single-core application
```

-----------------------

In single core application the single SHIELD parameter needs to be passed to the build command.
For example for nRF52840DK paired with nRF21540EK it will be:

    .. code-block:: console

       west build -b nrf52840dk_nrf52840 -- -DSHIELD=nrf21540_ek

> Multicore application

Reviewer 3: ```suggestion
Multi-core application
```

---------------------

> In multiple core application, where bluetooth runs on other than application core both cores needs to know about the FEM shield implemented.

Reviewer: Bluetooth Controlller ? The host still runs on the app core

Author: This comment is obsolete as whole sentence was removed. Now we are rely on the linked documentation.
